mypyc/genops.py

```diff
@@ -398,12 +426,21 @@ def prepare_class_def(module_name: str, cdef: ClassDef, mapper: Mapper) -> None:
             assert node.node.impl
             prepare_method_def(ir, module_name, cdef, mapper, node.node.impl)
 
-    # Special case exceptions and dicts
-    # XXX: How do we handle *other* things??
-    if info.bases and any(cls.fullname() == 'builtins.Exception' for cls in info.mro):
-        ir.builtin_base = 'PyBaseExceptionObject'
-    if info.bases and any(cls.fullname() == 'builtins.dict' for cls in info.mro):
-        ir.builtin_base = 'PyDictObject'
+    # Check for subclassing from builtin types
+    for cls in info.mro:
+        # Special case exceptions and dicts
+        # XXX: How do we handle *other* things??
+        if cls.fullname() == 'builtins.Exception':
+            # don't do anything for Exception (we'll do something for BaseException)
+            pass
+        elif cls.fullname() == 'builtins.BaseException':
+            ir.builtin_base = 'PyBaseExceptionObject'
+        elif cls.fullname() == 'builtins.dict':
+            ir.builtin_base = 'PyDictObject'
+        elif cls.fullname() == 'builtins.object':
+            pass
+        elif cls.fullname().startswith('builtins.'):
```

Comment:
What if the base class is a C class that is not defined in the builtins? At least add a comment.